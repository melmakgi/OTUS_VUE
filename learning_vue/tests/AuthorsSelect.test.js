import {describe, it, expect} from "vitest";
import Component from "@/components/AuthorsSelect.vue";
import {shallowMount, mount} from '@vue/test-utils'

const items = [{id: 1}, {id: 2}, {id: 3}, {id: 4}, {id: 5}];
const wrapper = mount(Component, {props: {items}});

describe('Component', () => {

    it('Mounts without errors', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.exists()).toBe(true);   // Проверка: нашли или нет
    })
    it("can simulate select selection", () => {
        const selected = wrapper.find("select");
        const options = selected.findAll("option");
        //const fooViewWrapper = shallowMount(items);
        //console.log(fooViewWrapper)
        selected.setValue("item 2");

        selected.trigger('input');
        // проверка, что событие было вызвано
        expect(wrapper.emitted()).toBeTruthy()
        console.log(wrapper.emitted());

        expect(selected.element.value);
        console.log(selected.element.value)
        options.at(2).element.selected = true
        console.log(selected.element.value)
    })
})