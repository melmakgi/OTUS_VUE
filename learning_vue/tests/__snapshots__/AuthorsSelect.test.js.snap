// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component > Renders All 1`] = `
"<select class=\\"form-select form-select-sm mb-3\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object]\\">
  <option disabled=\\"\\">Выберите автора для просмотра его книг</option>
  <option value=\\"Толстой Лев Николаевич\\">Толстой Лев Николаевич</option>
  <option value=\\"Иванов Алексей Викторович\\">Иванов Алексей Викторович</option>
</select>"
`;
